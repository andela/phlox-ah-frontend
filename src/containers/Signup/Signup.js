import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Modal, Row, Col
} from 'react-materialize';
import '../Common/ModalForm.scss';
import './Signup.scss';
import { connect } from 'react-redux';
import { signup } from '../../requests/SignupRequests';
import {
  Input, Button
} from '../BasePath';

/**
 *
 *
 * @class Signup
 * @extends {Component}
 */
class Signup extends Component {
  /**
   * @constructor function
   * @param {*} props React properties
   */
  constructor() {
    super();
    this.state = {
      email: '', username: '', password: '', confirmPassword: ''
    };
    this.onChange = this.onChange.bind(this);
    this.submit = this.submit.bind(this);
    this.onLoginClicked = this.onLoginClicked.bind(this);
  }

  /**
   * @memberOf handle close modal
   * @method closeModal
   * @param {*} event
   * @return {*} boolean
   */
  closeModal() {
    $('#signupModal').modal('close');
  }

  /**
   * @memberOf handleChangeEvent
   * @method handleChangeEvent
   * @param {*} event
   * @return {*} setstate
   */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  /**
   * @description - This method makes signup request
   * @param {objecj} event
   * @returns {object} null
   * @memberof Signup
   */
  submit(event) {
    event.preventDefault();
    this.props.signup(this.state);
  }


  /**
   * @description - This method runs when signup is clicked
   * @returns {object} null
   * @memberof Login
   */
  onLoginClicked() {
    this.closeModal();
    $('#login-modal').modal('open');
  }

  /**
   *
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   * @memberof Signup
   */
  render() {
    const {
      email, username, password, confirmPassword
    } = this.state;
    const { loading, success } = this.props;
    return (
      <Modal id='signupModal' className="center-align modal">
        {
          success
          && <div className="signupSuccessCard">
            <i className="fas fa-check-circle fa-5x"></i>
            <h5>Registration was successful</h5>
            <p>
              To proceed, Please check your email and verify your account
            </p>
            <Button
              waves='light'
              onClick={this.closeModal}
              name="OK"
            />
          </div>
        }
        {
          !success
          && <div>
            <div>
              <a
                className="close-modal"
                href="#" onClick={this.closeModal}>
                <i className="fas fa-times fa-lg black-text"></i>
              </a>
            </div>
            <h5 className="form-title">Authors Haven</h5>
            <form className="col s12" onSubmit={this.submit}>
              <Row>
                <Input
                  type="text"
                  label="Username"
                  s={12}
                  name="username"
                  id="username"
                  value={username}
                  onChange={this.onChange}
                  required={true}
                />
                <Input
                  type="email"
                  label="Email"
                  s={12}
                  name="email"
                  id="email"
                  value={email}
                  onChange={this.onChange}
                  required={true}
                />
                <Input
                  type="password"
                  label="Password"
                  s={12}
                  name="password"
                  id="password"
                  value={password}
                  onChange={this.onChange}
                  required={true}
                />
                <Input
                  type="password"
                  label="Confirm Password"
                  s={12}
                  name="confirmPassword"
                  id="confirmPassword"
                  value={confirmPassword}
                  onChange={this.onChange}
                  required={true}
                />
                <Button
                  type="submit"
                  className="signupButton"
                  waves='light'
                  name={loading
                    ? (<i className="fas fa-spinner fa-pulse"></i>)
                    : 'Sign Up'}
                />
              </Row>
            </form>
            <div className="or-divider">
              <span className="theme-color or">OR</span>
            </div>
            <h6 className="alt-label">SIGN UP USING</h6>
            <Row>
              <Col s={4} >
                <a
                  className="social-auth fb" href="#">
                  <i className="fab fa-facebook-f"></i>
                </a>
              </Col>
              <Col s={4}>
                <a className="social-auth gp "href="#">
                  <i className="fab fa-google-plus-g"></i>
                </a>
              </Col>
              <Col s={4}>
                <a className="social-auth tw" href="#">
                  <i className="fab fa-twitter"></i>
                </a>
              </Col>
            </Row>
            <div className="more-action">
              ALREADY HAVE AN ACCOUNT?
              <a href="#" onClick={this.onLoginClicked}>
                <span className="theme-color">
                  &nbsp; LOGIN
                </span>
              </a>
            </div>
          </div>
        }
      </Modal>
    );
  }
}

Signup.propTypes = {
  signup: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  success: PropTypes.bool,
};

const mapStateToProps = state => ({
  loading: state.Signup.loading,
  success: state.Signup.success,
});

export default connect(mapStateToProps, { signup })(Signup);
