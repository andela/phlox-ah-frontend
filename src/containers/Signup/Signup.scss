#signupModal{
    width:340px;
    input{
        margin-bottom:15px;
    }
    .input-field{
        margin:0em;
    }
    .modal-footer{
        display:none;
    }
    .fa-facebook{
        color:#4267B2;
    }
    .fa-google-plus-square{
        color:#CD4539;;
    }
    .fa-twitter-square{
        color:#00ACED;;
    }
    .theme-color{
        color:#06576D;
    }
    #signupButton{
        width:100%;
        border-radius: 25px;
        background-color: #06576D;
    }
    h5, h6{
        font-weight: bold;
    }
    .mb-2{
      margin-bottom: 0.5rem;
    }
    #loginSuccessCard{
        i{
            color:#3CBF98;
            margin-bottom:10px;
        }
        h6{
            color:#3CBF98;
        }
        button{
            border-radius: 20px;
            width:100%;
            height:45px;
            background:#3CBF98;
            margin-top: 10px;
            margin-bottom: 20px;
        }
    }
}
@media only screen and (min-width: 601px) {
    #signupModal{
        width:360px;
    }
}
@media only screen and (min-width: 993px) {
    #signupModal{
        width:380px;
    }
}
