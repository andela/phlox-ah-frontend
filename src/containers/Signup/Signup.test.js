import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import Signup from './Signup';

const mockStore = configureMockStore();
const store = mockStore({
  signup: {
    loading: true,
    success: false,
    failure: false,
    errors: []
  }
});

let component;
let myComponent;

describe('<Signup/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <Signup />
      </Provider>
    );
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have a form tag with class col', () => {
    expect(myComponent.find('form').hasClass('col')).toBe(true);
  });
  it('should have a signup button', () => {
    expect(myComponent.find('Button').exists()).toBe(true);
  });
  it('should have a input field', () => {
    expect(myComponent.find('Input').exists()).toBe(true);
  });
});
