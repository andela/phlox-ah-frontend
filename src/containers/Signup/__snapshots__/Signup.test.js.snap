// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> should render without throwing an error 1`] = `
<Connect(Signup)
  info={
    Object {
      "message": Array [
        "password is required",
      ],
      "success": false,
    }
  }
/>
`;
